server/src/main/java/io/crate/expression/symbol/Function.java

```diff
@@ -419,7 +419,7 @@ private void printOperator(StringBuilder builder, Style style, String operator)
     private void printSubscriptFunction(StringBuilder builder, Style style) {
         Symbol base = arguments.get(0);
         if (base instanceof Reference ref && base.valueType() instanceof ArrayType && ref.column().path().size() > 0) {
-            builder.append(ref.column().name());
+            builder.append(ref.column().getRoot().quotedOutputName());
```

Comment:
`printSubcriptFunction` is part of `toString(Style style)` which should prefer quoted version over non-quoted - https://github.com/crate/crate/blob/fded798afe8c7ced7bb1c97c8dfb17bbe340cf7a/server/src/main/java/io/crate/expression/symbol/Symbol.java#L79-L83